plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -380,6 +380,20 @@ private function update_pagination_anchors( $processor, $class_name, $key_prefix
 			if ( in_array( $class_name, array( 'wp-block-query-pagination-next', 'wp-block-query-pagination-previous' ), true ) ) {
 				$processor->set_attribute( 'data-wc-watch', 'callbacks.prefetch' );
 				$processor->set_attribute( 'data-wc-on--mouseenter', 'actions.prefetchOnHover' );
+			} else {
+				$inner_text = __( 'Page', 'woocommerce' );
+				$processor->set_bookmark( 'page_number' );
+				while ( $processor->next_token() ) {
+					if ( '#text' === $processor->get_token_type() ) {
+						$page = trim( $processor->get_modifiable_text() );
+						if ( is_numeric( $page ) ) {
+							$inner_text .= ' ' . $page;
+							break;
+						}
+					}
+				}
+				$processor->seek( 'page_number' );
+				$processor->set_attribute( 'aria-label', $inner_text );
```

Comment:
While it works as expected I don't think that's the way to go. Product Collection block uses `core/query-pagination` block underneath. Instead of iterating over and amending the HTML through Product Collection, I'd rather make updates in Gutenberg (and changes will eventually propagate to Product Collection block).

So [here in the source](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/query-pagination-numbers/index.php#L50), the changes could be done in the same way as in `plugins/woocommerce/templates/loop/pagination.php`. WDYT?

Reply:
@kmanijak while I don't 'disagree', I think that this is a good fix at the moment until we can make that update with Gutenberg's native pagination.

Reply:
> I think that this is a good fix at the moment until we can make that update with Gutenberg's native pagination.

I'm still against patching Gutenberg's block from within Product Collection for a couple of reasons:
1. Upstream contribution would provide improvement to a wider audience (Query Loop, Product Collection, Products (Beta), and any other block that may be using core's Pagination), not only to Product Collection users.
2. While the fix is provided in the Pagination block, this piece of code needs to be removed, otherwise the `aria-label` will be overridden.

Having said that, I'm happy to help with or even kick off a contribution to Gutenberg but at the same time I'm against introducing this code to Product Collection block. 🙌